src/token.rs

```diff
@@ -53,13 +72,19 @@ mod tests {
 
         let token: String = generate_token(&my_claims);
 
-        let token_data = match decode_token(&token) {
-            Ok(c) => c,
-            Err(err) => match *err.kind() {
-                ErrorKind::InvalidToken => panic!("Token is invalid"), // Example on how to handle a specific error
-                ErrorKind::InvalidIssuer => panic!("Issuer is invalid"), // Example on how to handle a specific error
-                _ => panic!("Some other errors"),
-            },
+        match verify_jwt_token(&token) {
```

Comment:
I was thinking may be I could do something like this:


let expect = String::from("Claims: Claims { sub: \"dummy_sub\", company: \"dummy_company\", iat: 1594146551, exp: 1594405751 }");

        let actual = match verify_jwt_token(&token) {
            Ok(value) => format!("Claims: {:?}", value.claims),
            Err(err) => format!("error: {}", err),
        };
        assert_eq!(expect,  actual);

Here, I want to store the return from match in  a variable actual (contains token data) as a string using format! so that i can compare it with expect using assert_eq! such that if both the strings are equal test will pass otherwise it will fail.

But what I found is very interesting value of fields iat and exp in actual comes out to be different than expect:

```
---- token::tests::test_token stdout ----
thread 'token::tests::test_token' panicked at 'assertion failed: `(left == right)`
  left: `"Claims: Claims { sub: \"dummy_sub\", company: \"dummy_company\", iat: 1594146551, exp: 1594405751 }"`,
 right: `"Claims: Claims { sub: \"dummy_sub\", company: \"dummy_company\", iat: 1594147079, exp: 1594406279 }"`', src/token.rs:81:9
```
and now I am confused how both can be different since we are decoding expect we should get the same value for all fields in decode part as they were in actual.

Reply:
huh. Ok, that is unexpected. It looks like the timestamps differ by 30 minutes or so. I don't see where `jsonwebtoken` might be creating new ones, so I'm not exactly sure where they might be coming from.
I know it's possible to decode a JWT manually. You take the second group, base64 decode it, and you wind up with JSON. It might be interesting to see if the dates are being altered by `generate` or `decode`.

Reply:
Ok, now I'm REALLY confused. I tried [running your code locally](https://gist.github.com/jrconlin/82ca1e6bf49f67dc71b8ac857484742c) and didn't get the same error you're seeing. I did a bit of clean-up on it too. 

I'm wondering if there might be something off about how the code or the operating system's reading your clock? The fact that the return is 30 minutes different may be a clue. I don't know if it's the problem, but I've seen things like this with other time sensitive code. 

I run my code in Unix, so I've tried running the tests as:
`TZ='Asia/Kolkata' RUST_LOG=debug cargo test -- --nocapture`
and got
```
    Finished test [unoptimized + debuginfo] target(s) in 0.17s
     Running target/debug/deps/tokenserver-538354d957e0f8bf

running 2 tests
I think it's local:2020-07-08T04:08:16.939730815+05:30 (UTC:2020-07-07T22:38:16.939773828Z)
test token::tests::test_invalid_token ... ok
Original: Claims { sub: "dummy_sub", company: "dummy_company", iat: 1594161496, exp: 1594420696 }
 decoded: Claims { sub: "dummy_sub", company: "dummy_company", iat: 1594161496, exp: 1594420696 }
test token::tests::test_token ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```
which matched up with my system clock. (I normally keep my servers on UTC)

Do you think that might be the problem? Is it possible for you to check your local clock or see if there's maybe something odd going on with how your system is reporting time?

Thanks!